reckoner/chart.py

```diff
@@ -243,28 +234,39 @@ def install(self, default_namespace=None, default_namespace_management={}, conte
             self._context = context
         # Try to run the install process for mark the result as failed
 
-        try:
-            self.manage_namespace()
+        # TODO: Improve error handling of a repository installation
+        #       Thoughts here, perhaps it would be better to install the
+        #       repositories *before* trying to install the chart. This
+        #       way we could find out earlier our course is wrong.
+        self.repository.install(self.name, self.version)
 
-            # Fire the pre_install_hook
-            self.pre_install_hook.run()
+        # Update the helm dependencies
+        self.update_dependencies()
 
-            # TODO: Improve error handling of a repository installation
-            #       Thoughts here, perhaps it would be better to install the
-            #       repositories *before* trying to install the chart. This
-            #       way we could find out earlier our course is wrong.
-            self.repository.install(self.name, self.version)
+        # Build the args for the chart installation
+        # And add any extra arguments
+        self.build_helm_arguments_for_chart()
 
-            # Update the helm dependencies
-            self.update_dependencies()
+        # Check and Error if we're missing required env vars
+        # TODO Rename this function as it does more than just "check", it also interpolates
```

Comment:
I think we can drop this TODO since we're probably not gonna do that (I know you were just carrying it over)